### PR TITLE
[Translation][cache fallback] keep only missing messages in fallbackCatalogue.

### PR DESCRIPTION
**Pull request by aitboudad**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Fixed tickets | #14315 |
| Tests pass? | yes |
| License | MIT |

### CODE REVIEW

File: src/Symfony/Component/Translation/Translator.php [hunk 1]
```diff
@@ -444,6 +430,14 @@ private function getFallbackContent(MessageCatalogue $catalogue)
             $fallbackSuffix = ucfirst(preg_replace($replacementPattern, '_', $fallback));
             $currentSuffix = ucfirst(preg_replace($replacementPattern, '_', $current));
 
+            $fallbackMessages = $fallbackCatalogue->all();
```
Comment (stloyd) [inline src/Symfony/Component/Translation/Translator.php]: This should be in `else` to avoid call if not needed as you replace it if not in debug anyway...

**Comment by mpdude** [comment]
I don't think this optimizes anything?

File: src/Symfony/Component/Translation/Translator.php (hunk 1 above)
Comment (aitboudad) [inline src/Symfony/Component/Translation/Translator.php]: fixed

**Comment by stof** [comment]
@mpdude this PR is about fixing a bug in the previous optimization. and keeping only the missing messages rather than the whole fallback catalogue is about optimizing memory usage.

**Comment by stof** [comment]
@nicolas-grekas we have a Fatal Error related to the Debug component in the deps=high build when running the TwigBundle tests. It would be great to investigate it

**Comment by mpdude** [comment]
@stof sure, I understand what it is about. The thing is that this optimization and dropping the messages causes unintuitive and possibly BC breaking behaviour when getting the catalogue from the translator (#14315).

As this PR passes the tests from #14315 I wonder if it *really* fixes the problem or if the actual optimization does not happen at all, although (at a glimpse) the code suggests it.

**Comment by aitboudad** [comment]
@mpdude I removed your last commit(e9cb79dd689e7d1a7417f498027beef88e8f7911) as It won't pass and I think it's useless

**Comment by mpdude** [comment]
No, it's not useless. It shows that your fix here does not suffice. Run it against a Symfony version before the optimization was introduced and it will pass.

**Comment by aitboudad** [comment]
@mpdude ok no problem for me :)

**Comment by mpdude** [comment]
I have an idea how we might be able to rescue this. It's not gonna be beautiful, but might work. Need to think about it a little more and will get back to you.

**Pull request closed without merging**